store/disk.go

```diff
@@ -46,64 +49,168 @@ func NewOnDiskStore(path string, pass []byte, opt ...Option) (Store, error) {
 	return store, nil
 }
 
-func (c *onDiskStore) Get(messageID imap.InternalMessageID) ([]byte, error) {
+const BlockSize = 64 * 4096
+
+func (c *onDiskStore) Set(messageID imap.InternalMessageID, in io.Reader) error {
+	if err := os.MkdirAll(c.path, 0o700); err != nil {
+		return err
+	}
+
 	if c.sem != nil {
 		c.sem.Lock()
 		defer c.sem.Unlock()
 	}
 
-	enc, err := os.ReadFile(filepath.Join(c.path, messageID.String()))
-	if err != nil {
-		return nil, err
+	nonce := make([]byte, c.gcm.NonceSize())
+	if _, err := rand.Read(nonce); err != nil {
```

Comment:
I've been using `crypto.RandomToken` in the past to have a "standardized" way of generating random tokens in bridge. I think under the hood they do the same thing as you, but to be safe, it might be good to stick to their implementation in case they improve it in future. Thoughts?

Reply:
I would address this in a follow MR to avoid scope creep :)